### PR TITLE
suggest using dynamic import instead of require

### PR DESCRIPTION
**Pull request by t6adev**
Hi, I just notice that here is used require().
Using import() instead is helpful in the doc.
Thank you.

### CODE REVIEW

File: docs/getting-started/integrate/browser.mdx [hunk 1]
```diff
@@ -77,6 +77,13 @@ import App from './App'
 if (process.env.NODE_ENV === 'development') {
   const { worker } = require('./mocks/browser')
   worker.start()
+  /*
+  // or you can do this instead
```
Comment (kettanaito) [inline docs/getting-started/integrate/browser.mdx:81]: The problem is that dynamic import works in a different way than `require` does, so they are not interchangeable in any regard. 

- `require` is not statically analyzable so bundlers cannot tree-shake its calls /  `import` can be tree-shaken.
- `require` is sync; `import` is async and must be awaited.

Those are very distinct differences and they directly affect how you should treat setups that utilize either method. 

On a related note, we're discussing an issue in Next where the usage of dynamic import results in a race condition between the worker activation and requests on the initial load (see https://github.com/vercel/next.js/issues/43284?notification_referrer_id=NT_kwDOAOSmz7M0OTA1NjI4NjY0OjE0OTg0OTEx&notifications_query=repo%3Avercel%2Fnext.js#issuecomment-1339929638).

**Review by kettanaito — COMMENTED** [review]
Hey, @t6adev. Thanks for proposing this. I've left a comment describing my concern with this suggestion.

**Comment by t6adev** [comment]
Thank you for your quick response.
I see, I didn't care about the race condition. And I missed the note below the code (also about ["Deferred mounting" page](https://mswjs.io/docs/recipes/deferred-mounting) )
My idea coming up while reading your comment is already described there 😓

Ok, now this PR is able to close.
I wonder if it can be helpful if there are some comments about the race condition and tree-shaking inside the example code block. I'm following your decision. Anyway, thank you again!

**Comment by kettanaito** [comment]
There may be an alternative route I'm experimenting with. With the Server Components in React, you can technically apply the interception and defer the client-side rendering even when using imports because async components are a thing now. Anybody reading this is free to experiment and share their findings.

**Pull request closed without merging**